### PR TITLE
Make sure proposal and saf instrument lists are unique

### PR DESCRIPTION
**Pull request by stuartcampbell**
Also revert recent 'fix' that removed the instrument list from the SAF

### CODE REVIEW

File: src/nsls2api/services/sync_service.py [hunk 1]
```diff
@@ -192,11 +192,11 @@ async def synchronize_proposal_from_pass(proposal_id: str) -> None:
         for resource in saf.Resources:
             beamline = await beamline_service.beamline_by_pass_id(str(resource.ID))
             if beamline:
-                beamline_list.append(beamline.name)
+                saf_beamline_list.append(beamline.name)
 
         saf_list.append(
             SafetyForm(
-                saf_id=str(saf.SAF_ID), status=saf.Status, instruments=saf_beamline_list
+                saf_id=str(saf.SAF_ID), status=saf.Status, instruments=set(saf_beamline_list)
```
Comment (padraic-shafer) [inline src/nsls2api/services/sync_service.py:199]: @JunAishima 
Is the order of this list important to consumers?

**Review by padraic-shafer — APPROVED** [review]
Fix looks good.  I left a question about whether order needs to be preserved, which can be dealt with later in case you need to merge this now.

**Review by JunAishima — APPROVED** [review]
looks good! as I mentioned in person, this should also fix the issue of duplicate beamline names in the instruments list for a proposal

File: src/nsls2api/services/sync_service.py (hunk 1 above)
Comment (stuartcampbell) [reply src/nsls2api/services/sync_service.py:199]: No, should not be.

**Pull request merged**